lib/ASRouter.jsm

```diff
@@ -223,6 +226,26 @@ class _ASRouter {
           newState.messages = [...newState.messages, ...messages];
         }
       }
+
+      // Some triggers have trigger listeners that we might need to initialise.
+      for (const [triggerID, listener] of ASRouterTriggerListeners) {
+        const triggerMessages = newState.messages
+          .filter(m => m.trigger && m.trigger.id === triggerID);
+
+        if (triggerMessages.length) {
+          // If some messages have this trigger, initialise the trigger listener
+          const params = new Set();
+          for (const message of triggerMessages) {
+            ((message.trigger.params) || []).forEach(p => params.add(p));
```

Comment:
I think you can literally just make a `new Set()` directly from params if that's what you're looking to do... no need to manually add each item: `new Set(params || [])`

Reply:
Clarified with Adam about the use of Set and multiple parameters for different messages with the same trigger. As discussed over Slack we'll take the following approach:
> iterate over the messages in ASRouter and just blindly send the params and call init to the listener, and then internally the listener would know that it already has a running list of hosts, and to just add these new params to the set it already has